src/Reindexer.h

```diff
@@ -4,6 +4,8 @@
 #include <functional>
 #include <vector>
 #include <algorithm>
+#include <boost/range/adaptor/map.hpp>
+#include <boost/range/algorithm/copy.hpp>
```

Comment:
Since you didn't end up using those, we don't need to include it.

Reply:
Opps, yes. Do you have an idea how to sort `std::pair<T, int>' by value in a stream? I couldn't find anything so far.

Reply:
Since we know that the `int` component is unique and dense, insertion is going to be the best solution.
We can still do a range-based for loop to make it look nicer.

Reply:
Ok, lets go for the range based loop then for now :smile:.